### ISSUE TITLE
[4.x] Access to Local Storage can be blocked in Chrome

### ISSUE THREAD

**Issue by olafcm** [algolia/algoliasearch-client-javascript#997]
We updated to v4 of the client.
Since then we started to receive the following error for Chrome users:

> Failed to read the 'localStorage' property from 'Window': Access is denied for this document.

The issue happens here:

```
function u(e) {
  var t = e.localStorage || window.localStorage,
    r = "algoliasearch-client-js-".concat(e.key),
    n = function() {
      return JSON.parse(t.getItem(r) || "{}")
    };
```

More info: https://www.chromium.org/for-testers/bug-reporting-guidelines/uncaught-securityerror-failed-to-read-the-localstorage-property-from-window-access-is-denied-for-this-document

Is it possible for the client to detect if the LocalStorage is available? I.e. use: https://github.com/Modernizr/Modernizr/blob/master/feature-detects/storage/localstorage.js

Thank you

**Comment by nunomaduro** [comment]
Thanks for reporting this issue, I will investigate this tomorrow morning. 👍

**Comment by nunomaduro** [comment]
@olafcm Just to be clear, your search is working, but you got the `warning` on the chrome console right?

**Comment by olafcm** [comment]
Yes, I tested it and search works. It probably only affects the search results caching.

**Comment by nunomaduro** [comment]
Actually just don't preserve the state of the hosts after the refresh: https://github.com/algolia/algoliasearch-client-javascript/blob/d862a912d764655a415a9bffc25634d625ebe88d/packages/algoliasearch/src/builds/browserLite.ts#L49.

I will see if we can avoid the warning on the console. Otherwise everything seems to be working as expected.

**Comment by Haroenv** [comment]
I wonder if adding a .catch which still throws, but without the default error message will avoid it showing on the console.

https://github.com/algolia/algoliasearch-client-javascript/blob/d862a912d764655a415a9bffc25634d625ebe88d/packages/cache-browser-local-storage/src/createBrowserLocalStorageCache.ts#L31

**Comment by nunomaduro** [comment]
We are using `catch` on the `FallbackableCache`:  https://github.com/algolia/algoliasearch-client-javascript/blob/d862a912d764655a415a9bffc25634d625ebe88d/packages/cache-common/src/createFallbackableCache.ts#L21

That performs a `catch` on the driver to fallback to the next one. Leet me investigate this.

**Comment by nunomaduro** [comment]
@olafcm can you also review the pull request that I just made?

**Comment by nunomaduro** [comment]
The bug has been fixed in version `4.0.2`. ⚡️

**Comment by olafcm** [comment]
Works perfect, thank you very much for the perfect and quick fix @nunomaduro 🎈
